src/dev/precommit_hook/check_lockfile_symlinks.js

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { dirname } from 'path';
+import { existsSync, lstatSync, readFileSync } from 'fs';
+
+import { createFailError } from '@kbn/dev-utils';
+import { matchesAnyGlob } from '../globs';
+
+import { LOCKFILE_GLOBS, MANIFEST_GLOBS, IGNORE_FILE_GLOBS } from './lockfile_check_config';
+
+function listPaths(paths) {
+  return paths.map(path => ` - ${path}`).join('\n');
+}
+
+async function checkProhibitedLockfiles(log, files) {
```

Comment:
```suggestion
async function checkOnlyLockfileAtProjectRoot(log, files) {
```
It took me a second to understand where we were prohibiting lockfiles, the suggested name isn't great...